server/src/database/delayedStorage.ts

```diff
@@ -0,0 +1,62 @@
+export class DelayedStorage<T> {
+  private dataStore: T | null
+  private readonly getRemote: () => T | null
+  private readonly saveNow: () => void
+  private readonly saveLater: () => void
+  private timeoutId: NodeJS.Timeout | null
+
+  constructor (
+    getRemote: () => T | null,
+    saveRemote: (value: T) => void,
+    timeout: number,
+    initialValue?: T,
+  ) {
+    this.timeoutId = null
+    this.getRemote = getRemote
+    this.saveNow = () => {
```

Comment:
Don't set this in the constructor.    Just put saveRemote in your properties, and declare a method